### PR TITLE
Fetch nfts

### PR DESCRIPTION
**Pull request by RabbitDoge**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

**Review by Chef-Cheems — APPROVED** [review]

**Review by ChefHutch — APPROVED** [review]

File: src/state/nftMarket/helpers.ts [hunk 1]
```diff
@@ -27,3 +27,27 @@ export const getCollections = async (): Promise<any[]> => {
     return []
   }
 }
+
+export const getNftsFromCollection = async (collectionAddress: string): Promise<any[]> => {
```
Comment (ChefKai) [inline src/state/nftMarket/helpers.ts:31]: The query will be faster if you fetch the `Collection` entity and fetch all subsequent tokens, such as:

```graphql
{
  collection(id: "${collectionAddress}") {
    id
    NFTs {
      currentSeller
      isTradable
      tokenId
      metadataUrl
    }
  }
}
```

Also, you need to make sure the `collectionAddress` is lowercased

File: src/state/nftMarket/types.ts [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { BigNumberish } from 'ethers'
 
 export interface State {
   data: {
-    collections: Collection[]
-    users: User[]
+    collections: { [key: string]: Collection }
```
Comment (0xCorgi) [inline src/state/nftMarket/types.ts:8]: We could use a Record here: `Record<string, Collection>`

File: src/state/nftMarket/helpers.ts [hunk 3]
```diff
@@ -27,3 +27,27 @@ export const getCollections = async (): Promise<any[]> => {
...
+      GRAPH_API_NFTMARKET,
+      gql`
+        {
+          nfts(where: { collection: "${collectionAddress}" }) {
+            id
+            currentSeller
+            isTradable
+            tokenId
+            metadataUrl
+          }
```
Comment (0xCorgi) [inline src/state/nftMarket/helpers.ts:43]: Should we put GraphQL queries in a separate folder ?

**Review by 0xCorgi — APPROVED** [review]
Looks good !
Just a few suggestions.

File: src/state/nftMarket/helpers.ts [hunk 4]
```diff
@@ -27,3 +27,27 @@ export const getCollections = async (): Promise<any[]> => {
...
+        }
+      `,
+    )
+
+    return res.nfts
+  } catch (error) {
+    console.error('Failed to fetch NFTs', error)
+    return []
+  }
+}
```
Comment (hachiojidev) [inline src/state/nftMarket/helpers.ts:53]: We want to pass variables via the second parameter of `gql`. Check `state/predictions/helpers.ts` line 257 for a similar example.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/state/nftMarket/helpers.ts (hunk 1 above)
Comment (ChefKai) [reply src/state/nftMarket/helpers.ts:31]: Please, also leverage the filter parameters included within graphql library such as for the Prediction: https://github.com/pancakeswap/pancake-frontend/blob/develop/src/state/predictions/helpers.ts#L257

File: src/state/nftMarket/helpers.ts (hunk 4 above)
Comment (hachiojidev) [reply src/state/nftMarket/helpers.ts:53]: https://github.com/pancakeswap/pancake-frontend/blob/a9d7070c6975f1a868516ba6a9f36acbfc54945d/src/state/predictions/helpers.ts#L257

File: src/state/nftMarket/helpers.ts (hunk 3 above)
Comment (RabbitDoge) [reply src/state/nftMarket/helpers.ts:43]: it's already in a separate file

File: src/state/nftMarket/helpers.ts (hunk 4 above)
Comment (RabbitDoge) [reply src/state/nftMarket/helpers.ts:53]: You mean like here https://github.com/pancakeswap/pancake-frontend/blob/develop/src/state/info/queries/tokens/tokenData.ts#L72 ? 😄

Comment (hachiojidev) [reply src/state/nftMarket/helpers.ts:53]:
```
export const getBet = async (betId: string): Promise<BetResponse> => {
  const response = await request(
    GRAPH_API_PREDICTION,
    gql`
      query getBet($id: ID!) {
        bet(id: $id) {
          ...
          round {
            ...
          }
          user {
            ...
          } 
        }
      }
  `,
    {
      id: betId.toLowerCase(),
    },
  )
  return response.bet
}
```
This is the simplest example I can find. Notice `id: betId.toLowerCase(),` is passed in as the second parameter and the query is wrapped.

File: src/state/nftMarket/helpers.ts [hunk 5]
```diff
@@ -20,10 +20,35 @@ export const getCollections = async (): Promise<any[]> => {
...
+            NFTs {
+              currentSeller
+              isTradable
+              tokenId
+              metadataUrl
+            }
+          }
+        }
+      `,
+      { collectionAddress },
```
Comment (hachiojidev) [inline src/state/nftMarket/helpers.ts:47]: As @ChefKai mentioned, you will want to make sure this is lowercase.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**